packages/cardano-services/test/ChainHistory/DbSyncChainHistoryProvider/ChainHistoryBuilder.test.ts

```diff
@@ -47,61 +50,64 @@ describe('ChainHistoryBuilder', () => {
 
   beforeEach(() => logger.reset());
 
-  describe('queryTransactionInputsByHashes', () => {
+  const hashesToId = async (hashes: Cardano.TransactionId[]) => {
```

Comment:
This can return IDs in different order. I suggest doing this and then if you need multiple you can do `Promise.all(arr.map(getTxId))`
```suggestion
  const getTxId = async (hash: Cardano.TransactionId) => {
```
Alternatively could return a Set or just add a docblock to make the behavior clear.

Reply:
Nice catch!
4cb75f987 should solve this